### PR TITLE
Add protected routes and video option button functionality

### PR DESCRIPTION
**Pull request by VLeads**
This pull request adds private and protected routes. 
Add an option button on video cards for add to watch later and add to cart functionality.
Add login guest button.

And some CSS fixes are also done.

### CODE REVIEW

File: src/components/card/HorizontalVideoCard.jsx [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+          >
+            <OptionsIcon />
+          </span>
+          {showVideoMenu && (
+            <ol className={`list ${styles.horizontalVideoMenuCard}`}>
+              <li
+                className={styles.videoMenuCardItem}
+                onClick={watchLaterClickHandler}
+              >
+                {watchLater.data.some((element) => element._id === _id) ? (
```
Comment (ShekharDhangar) [inline src/components/card/HorizontalVideoCard.jsx:132]: you could make this a function and use it everywhere

File: src/components/modal/Modal.js [hunk 2]
```diff
@@ -66,16 +64,18 @@ export function Modal() {
...
           {playlist?.data.length > 0 &&
+            !notShowPlaylistNames &&
             playlist?.data.map((item) => (
               <div key={item._id}>
                 <label className="playlist_label">
                   <input
                     type="checkbox"
-                    checked={item?.videos.find(
+                    id={item.title}
+                    checked={item?.videos.some(
```
Comment (ShekharDhangar) [inline src/components/modal/Modal.js:74]: same you could make the some method check into a function and use it everywhere

File: src/pages/liked/Liked.jsx [hunk 3]
```diff
@@ -37,7 +37,6 @@ function Liked() {
             style={{
               textDecoration: "underline",
               color: "red",
```
Comment (ShekharDhangar) [inline src/pages/liked/Liked.jsx:39]: avoid using inline-styles

File: src/pages/playlist/Playlist.jsx [hunk 4]
```diff
@@ -1,38 +1,57 @@
...
+  const { getToken: authToken } = useUser();
+  const { isModalOpen, setIsModalOpen, setNotShowPlaylistNames } = useToast();
+  const { deletePlaylistNameApi } = useApi();
 
   const { playlist, deleteDataUsingApi } = usePlaylist();
   const { data: playlistData } = playlist;
 
-  const { deletePlaylistNameApi } = useApi();
-
   const currentPlaylist = playlistData.find((item) => item._id === playlistId);
```
Comment (ShekharDhangar) [inline src/pages/playlist/Playlist.jsx:20]: here also you could make a findItem function and use it everywhere

**Review by ShekharDhangar — APPROVED** [review]
Apart from suggested changes. rest looks good

File: src/pages/playlist/Playlist.jsx [hunk 5]
```diff
@@ -1,38 +1,57 @@
...
+    }
+  };
+
   return (
     <div className="history_videos_container playlist_container">
       <div className="playlist-sidebar">
         <div className="page_heading playlist_heading">Playlists</div>
+        <button
+          className="btn btn-primary btn-primary-outline"
+          style={{ marginBottom: "2rem" }}
```
Comment (Developers-codz) [inline src/pages/playlist/Playlist.jsx:45]: avoid inline css

**Review by Developers-codz — APPROVED** [review]
Good work on code. Nothing much to suggest. Everything looks good to go.

**Comment by VLeads** [comment]
Merging. Thanks for the review

**Pull request merged**